scripts/Fig6-FigS7_bulk-analysis.R

```diff
@@ -27,10 +38,81 @@ geneset_files <- list.files(
     \(x) {stringr::str_split_i(basename(x), pattern = "_", i = 1)}
   )
 
-sample_metadata_file <- here::here("s3_files", "scpca-library-metadata.tsv")
+# The TSV files containing odds ratio results from overrepresentation analysis
+odds_files <- list.files(
+  path = result_dir,
+  pattern = "_ORA-odds-ratios\\.tsv$",
+  full.names = TRUE
+)|>
+  purrr::set_names(
+    \(x) {stringr::str_split_i(basename(x), pattern = "_", i = 1)}
+  )
 
 
-# First, create the manuscript-numbers table ----------------------------------------------------------------
+# The TSV files containing modeled values
+data_files <- list.files(
+  path = file.path(result_dir, "models"),
+  pattern = "-data_threshold-0\\.tsv$",
+  full.names = TRUE
+)|>
+  purrr::set_names(
+    \(x) {stringr::str_split_i(basename(x), pattern = "_", i = 1)}
+  )
+
+
+# Output files
+bulk_numbers_tsv <- file.path(tables_dir, "bulk-analysis-counts.tsv")
+main_pdf <- file.path(pdf_dir, "Fig6_bulk-results.pdf")
+si_pdf <- file.path(pdf_dir, "FigS8_bulk-results.pdf")
+
+# Functions --------------------------------------------------------------------
+
+
+# Function to plot the bulk ~ pseudobulk scatterplot panels
+plot_scatterplot <- function(df) {
+  ggplot(df) + 
+    aes(x = pseudobulk, y = bulk) + 
+    geom_bin_2d(binwidth = 0.25) + 
+    scale_fill_viridis_c(limits = c(0, 800), oob = scales::squish, option = "inferno") +
+    geom_smooth(method = "lm", color = "paleturquoise", linewidth = 0.75) + 
+    facet_wrap(vars(project_facet)) +
+    labs(
+      x = "Pseudobulk expression", 
+      y = "Bulk expression", 
+      fill = "Point density"
+    )
+}
+
+# Function to plot the odds ratio barplot panels
+plot_odds_ratios <- function(df) {
+  ggplot(df) +
+    aes(
+      x = tidytext::reorder_within(geneset_cell_type, odds_ratio, within = project_facet),
+      y = odds_ratio, 
+      fill = -log10(p_adj_bh)
+    ) +
+    geom_col() +
+    geom_text(
+      aes(
+        label = round(odds_ratio, 2), 
+        y = odds_ratio + 0.75
+      ), 
+      size = 2.5
+    ) +
+    scale_fill_viridis_c() +
+    tidytext::scale_x_reordered() + # gets the labels back to only geneset_cell_type
+    facet_wrap(vars(project_facet), scales = "free_x", nrow = 1) +
+    labs(
+      x = "Consensus cell type", 
+      y = "Odds ratio", 
+      fill = "-Log10(P)"
```

Comment:
Might as well be a bit more verbose. We seem to have the room!
```suggestion
      fill = "-Log10(adj. p-value)"
```

Reply:
done